src/algebra/lie_algebra.lean

```diff
@@ -205,16 +209,191 @@ def of_associative_algebra (A : Type v) [ring A] [algebra R A] : lie_algebra R A
 An important class of Lie algebras are those arising from the associative algebra structure on
 module endomorphisms.
 -/
-def of_endomorphism_algebra (M : Type v)
+instance of_endomorphism_algebra (M : Type v)
   [add_comm_group M] [module R M] : lie_algebra R (module.End R M) :=
-of_associative_algebra R (module.End R M)
+of_associative_algebra (module.End R M)
+
+/--
+The adjoint action of a Lie algebra on itself.
+-/
+def Ad (x : L) : L →ₗ[R] L :=
```

Comment:
Should this be redefined as a Lie alg morphism from `L` to `module.End R M`?

Reply:
You do that in `bil_lie_morphism` but that name is quite verbose. I think people wouldn't mind having `Ad` available for the adjoint representation.

Reply:
I'm not absolutely sure what you have in mind here; would the following capture it:

- rename `Ad` to `Ad_fun`,
- kill `bil_lie` by merging into what is currently called `bil_lie_morphism`,
- rename `bil_lie_morphism` to `Ad`?

Reply:
Yes, I think that sounds good. (Sorry for the delay, I somehow missed this question.)

Reply:
I think I've cleaned up the mess here now :-)